stdlib/src/collections/dict.mojo

```diff
@@ -478,6 +478,14 @@ struct Dict[K: KeyElement, V: CollectionElement](
         self._index = existing._index^
         self._entries = existing._entries^
 
+    fn clear(inout self):
+        """Remove all elements from the dictionary."""
+        self.size = 0
+        self._n_entries = 0
+        self._reserved = 8
```

Comment:
This number `8` can also be found in the `__init__` function. Maybe it would make sense to group them and have them accessible as an alias? Something like `Self._inital_reservation` or whatever name you prefer.